Sources/Apodini/Application/Application+Connection.swift

```diff
@@ -1,21 +1,18 @@
 //
 //  Connection.swift
-//  
+//
 //
 //  Created by Moritz Schüll on 09.12.20.
 //
 
 import Foundation
 
-enum ConnectionEnvironmentKey: EnvironmentKey {
-    static var defaultValue = Connection()
-}
-
-extension EnvironmentValues {
+extension Application {
     /// A Property identifying the `Connection` which provides an abstract view on the underlying protocol's state.
+    /// - Note: The setter should never be directly called and only serves as a placeholder.
     public var connection: Connection {
-        get { self[ConnectionEnvironmentKey.self] }
-        set { self[ConnectionEnvironmentKey.self] = newValue }
+        get { Connection() }
```

Comment:
I would even suggest to add a `fatalError` for accessing a `Connection` on the `Application` level should not possible as this is stored on a `Environment` property wrapper level. 

Thinking about this now we could reconsider adding back the `EnvironmentKeys` as indicators that they are only stored in the `@Environment` property wrapper's key value storage and accessing them does not even take a look at the `Application` storage while accessing an environment value using an `Application` key path first checks the storage of the single `@Environment` property wrapper and then the `Application` storage?

Reply:
I think the idea was to have a default `Connection` with the state `.end`. But I think the default case is also handled by all exporters, so changing this the getter to a `fatalError` shouldn't change anything. Maybe @moritzschuell can have a look at this.

I don't think it is necessary to re-introduce `EnvironmentKeys`. The behavior of `@Environment` is to always check `dynamicProperties` first and only if it is missing then to check the `Application` storage.

Reply:
Sounds good, I think that handling this on the exporter level or ideally on the `SemanticModelLevel` sounds good to me.
Ok, makes sense 👍

Reply:
@awoc Yes, you're right. The `ConnectionContext` will make sure to always supply a `Connection` to the handler, with state `.end` by default. So, an application level `Connection` access should never happen (and also doesn't really make sense semantically).